api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -98,7 +97,235 @@
 	public void runBeforeEachTest() throws Exception {
 		executeDataSet(ENC_INITIAL_DATA_XML);
 	}
-	
+
+	@Test
+	@Verifies(value = "should save encounter correctly when a leaf child obs updated without recreating an entire obs hierarchy", method = "saveEncounter(Encounter)")
+	public void saveEncounter_shouldNotCreateACopyOfParentObsButEditChildObs(){
+		ConceptService cs = Context.getConceptService();
+		EncounterService es = Context.getEncounterService();
+		ObsService os = Context.getObsService();
+
+		//create an obs
+		Obs o = createObs(cs.getConcept(3),"original obs value text");
+		//create a second obs
+		Obs o2 = createObs(cs.getConcept(3),"second obs value text");
+		//create a parent obs
+		Obs oParent =createObs(cs.getConcept(23), null); //in the concept set table as a set
+
+		//add o and o2 to the parent obs
+		oParent.addGroupMember(o2);
+		oParent.addGroupMember(o);
+
+		Encounter enc = buildEncounter();
+		enc.addObs(oParent);
+
+		enc = es.saveEncounter(enc);
```

Comment:
Don't you have to flush the session for the test to be concrete, couldn't you actually use an existing encounter already in the DB?